### PR TITLE
Fix edm4hep output for DDG4

### PR DESCRIPTION
**Pull request by gaede**
BEGINRELEASENOTES
- finalize the EDM4hep output module for DDG4/ddsim
      - add treatment of event parameter meta data

ENDRELEASENOTES

### CODE REVIEW

**Comment by vvolkl** [comment]
Looks good to me. As far as I can tell, this needs `DDG4/EventParameters.h` and therefore the current DD4hep master. Should we merge this anyway already?

File: plugins/DDG4EDM4hep/edm4hep/Geant4Output2EDM4hep.cpp [hunk 1]
```diff
@@ -20,25 +20,28 @@
...
-//#include "EDM4hepEventParameters.h"
 #include "podio/EventStore.h"
 #include "podio/ROOTWriter.h"
 
 #include <typeinfo>
 #include <iostream>
+#include <ctime>
 
-using namespace edm4hep ;
+//using namespace edm4hep ;
```
Comment (vvolkl) [inline plugins/DDG4EDM4hep/edm4hep/Geant4Output2EDM4hep.cpp:44]:
```suggestion

```

File: plugins/DDG4EDM4hep/edm4hep/Geant4Output2EDM4hep.cpp [hunk 2]
```diff
@@ -366,35 +380,36 @@ void Geant4Output2EDM4hep::saveEvent(OutputContext<G4Event>& ctxt)  {
...
+    runNumber = parameters->runNumber() + runNumberOffset;
+    eventNumber = parameters->eventNumber() + eventNumberOffset;
+    parameters->extractParameters(*m_store);
+  } else { // ... or from DD4hep framework
+    runNumber = m_runNo + runNumberOffset;
+    eventNumber = ctxt.context->GetEventID() + eventNumberOffset;
+  }
   printout(INFO,"Geant4Output2EDM4hep","+++ Saving EDM4hep event %d run %d.",
-     eventNumber, runNumber);
+	   eventNumber, runNumber);
```
Comment (andresailer) [inline plugins/DDG4EDM4hep/edm4hep/Geant4Output2EDM4hep.cpp:397]: Please, no "tabs".

File: plugins/DDG4EDM4hep/edm4hep/Geant4Output2EDM4hep.cpp [hunk 3]
```diff
@@ -366,35 +380,36 @@ void Geant4Output2EDM4hep::saveEvent(OutputContext<G4Event>& ctxt)  {
...
 
-  //FIXME: need a suitable EventHeader object in EDM4hep
-  //        and a way to store meta data parameters with this
+  // this does not compile as create() is we only get a const ref - need to review PODIO EventStore API
```
Comment (vvolkl) [inline plugins/DDG4EDM4hep/edm4hep/Geant4Output2EDM4hep.cpp:399]: This would be fixed https://github.com/AIDASoft/podio/pull/95

**Pull request merged**